reviewers.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+import operator
+import os
+import re
+
+
+def prompt_for_user():
+    while True:
+        try:
+            user_input = input("\nName or email (case insensitive): ") 
+        except (KeyboardInterrupt, EOFError):
+            return None
+        clean_input = user_input.strip().lower()
+        if clean_input != "":
+            return clean_input
+
+
+if __name__ == "__main__":
+    print("Utility to help generate 'Reviewers' string for Pull Requests. Use Ctrl+D or Ctrl+C to exit")
+    
+    stream = os.popen("git log | grep Reviewers")
+    lines = stream.readlines()
+    all_reviewers = defaultdict(int)
+    for line in lines:
+        stripped = line.strip().lstrip("Reviewers: ")
+        reviewers = stripped.split(",")
+        for reviewer in reviewers:
+            all_reviewers[reviewer.strip()] += 1 
+    parsed_reviewers = []
+
+    for item in all_reviewers.items():
+        m = re.match("(?P<name>.*)\s<(?P<email>.*)>", item[0])
+        if m is not None and len(m.groups()) == 2:
+            if item[1] > 2:
+                parsed_reviewers.append((m.group("name"), m.group("email"), item[1]))
+    
+    selected_reviewers = []
+    while True:
+        if len(selected_reviewers) != 0:
+            print(f"Reviewers so far: {selected_reviewers}")
+        user_input = prompt_for_user()
+        if user_input is None:
+            break
+        candidates = []
+        for reviewer, email, count in parsed_reviewers:
+            if reviewer.lower().startswith(user_input) or email.lower().startswith(user_input):
+                candidates.append((reviewer, email, count))
+            if len(candidates) == 10:
+                break
+        if len(candidates) == 0:
+            continue
+
+        print("\nPossible matches (in order of most recent):")
```

Comment:
Should we mention that the number in the parenthesis is the number of occurrences? How about something like:
```python
        print("\nMatches order by most recent (the number in the parenthesis is the number of occurrences):")
```
or we can change the formatting to something like:
```
Name <email> (reviews=123)
```

Reply:
Should we mention we only list top 10 candidates?

Reply:
@showuon the zip + range(10) has the effect of only showing the top 10